core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -3570,6 +3591,72 @@ public long generateSeqNo() {
         }
     }
 
+    public void testMinGenerationForSeqNo() throws IOException, BrokenBarrierException, InterruptedException {
```

Comment:
The reason that I added a test at this level is to try to test that when the engine flushes it is pushing the right generations to the translog commit.